### PR TITLE
feat: define scroll speed + smart_send to qflist

### PR DESCRIPTION
**Pull request by elianiva**
continuing @Conni2461's work #606

you can now do something like this to customise the scrolling speed, the default is `preview_height / 2`

```lua
layout_defaults = {
  horizontal = {
    scroll_speed = 5
  },
  vertical = {
    scroll_speed = 2
  },
}
```

### CODE REVIEW

File: lua/telescope/actions/init.lua [hunk 1]
```diff
@@ -143,14 +143,21 @@ function actions.toggle_selection(prompt_bufnr)
   current_picker:toggle_selection(current_picker:get_selection_row())
 end
 
+local get_scroll_speed = function(prompt_bufnr)
```
Comment (Conni2461) [inline lua/telescope/actions/init.lua:146]: I am thinking maybe we should move this to `action_set`. Similar to `set.shift_selection = function(prompt_bufnr, change)` and all we need to pass in is direction (-1 or 1)

Comment (Conni2461) [reply lua/telescope/actions/init.lua:146]: Otherwise LGTM. Thanks for picking it up :)

File: lua/telescope/actions/set.lua [hunk 2]
```diff
@@ -124,6 +125,18 @@ set.edit = function(prompt_bufnr, command)
   end
 end
 
+--- Scrolls the previewer up or down
+---@param prompt_bufnr number: The prompt bufnr
+---@param direction number: The direction of the scrolling
+--      Valid directions include: 1 or -1
```
Comment (Conni2461) [inline lua/telescope/actions/set.lua:131]: FYI Multiline param currently not possible with `tree-sitter-lua` See https://github.com/tjdevries/tree-sitter-lua/pull/14#discussion_r587908659

**Comment by Conni2461** [comment]
2 things about action docs:
1. There is this: https://github.com/nvim-telescope/telescope.nvim/pull/587 I just don't wanna take away his first PR. Sorry i know you spend time documenting `actions/init.lua`. We can keep the `set.lua` documentation :)
2. We have the max query issue until neovim is using tree sitter 0.19.0 and tj regenerated grammar See https://github.com/nvim-treesitter/nvim-treesitter/issues/963

**Comment by elianiva** [comment]
sure, I can remove them. I didn't spend that much time, it's just a lot of copypaste because I was thinking *why not?* :laughing:

File: lua/telescope/actions/set.lua (hunk 2 above)
Comment (elianiva) [reply lua/telescope/actions/set.lua]: I should try that grammar as well, been wanting to try that but I always forgot :D

Comment (Conni2461) [reply lua/telescope/actions/set.lua]: Wouldn't recommend it quite yet. I am using a quite out of date `nvim-treesitter` to make it work.
I would wait until overriding a query is easier in `nvim-treesitter` and the queries for that grammar need some work too.

The only thing that you would gain from it is the emmy documentation part and you can play with `docgen` which is super cool :)

**Review by Conni2461 — APPROVED** [review]
LGTM

Thanks for picking it up :)

**Pull request merged**